app/helpers/assets_helper.rb

```diff
@@ -48,40 +48,36 @@ def instance_variable_or_id_param(name, &block)
 
   # Returns a select tag that has it's options ordered by name (assumes present of sorted_by_name function)
   # and disabled if a value has been pre-selected.
-  def select_field_sorted_by_name(field, select_options_source, selected, options = {})
+  def select_field_sorted_by_name(field, select_options_source, selected, can_edit, options = {})
+    disabled = selected.present? && !can_edit
+
     tag.div(class: 'col-md-5') do
       select_tag(
         field,
         options_for_select(select_options_source.sorted_by_name.pluck(:name, :id), selected.try(:to_i)),
-        options.merge(disabled: (selected.present? and not current_user.administrator?),
-                      class: 'form-control select2')
+        options.merge(disabled: disabled, class: 'form-control select2')
       )
     end
   end
 
   # Returns true if the current user can request additional sequencing on the given asset, otherwise false
   def current_user_can_request_additional_sequencing_on?(asset)
-    asset.sequenceable? && # Asset must be sequenceable ...
-      (current_user.administrator? || # ... user could be an administrator ...
-        current_user.manager?) # ... or a manager
+    asset.sequenceable? && can?(:create_additional, Request)
   end
 
   # Returns true if the current user can request an additional library on the asset, otherwise false
   def current_user_can_request_additional_library_on?(asset)
-    asset.is_a?(SampleTube) && current_user.administrator?
+    asset.is_a?(SampleTube) && can?(:create_additional, Request)
   end
```

Comment:
Serious contemplated nuking this, as it doesn't really work any more, or make sense when it comes to the current submission system